### ISSUE TITLE
Proposal: Command/Query Separation for ROM::SQL::Commands

### ISSUE THREAD

**Issue by Krule** [rom-rb/rom-sql#257]
According to [CQS principle](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation): 

> Every method should either be a **command** that performs an action, or a **query** that returns data to the caller, but not both.

With that in mind I would like to see a refactor of the following with at least an option for a commands not to perform additional database query in order to return the updated state (especially since we usually know what state is expected to be a result of the execution in advance):

  - https://github.com/rom-rb/rom-sql/blob/master/lib/rom/sql/commands/create.rb
  - https://github.com/rom-rb/rom-sql/blob/master/lib/rom/sql/commands/update.rb
  - https://github.com/rom-rb/rom-sql/blob/master/lib/rom/sql/commands/delete.rb

**Comment by abrthel** [comment]
@solnic you were saying that you regret adding `result: :many` or `result: :one` but maybe a custom command set with `result: :void` could trigger this behavior?

**Comment by solnic** [comment]
@abrthel oh we definitely do not want to add another option for `result` :) I *think* all that's needed, is splitting methods in a way that we don't get results back in the same method. Once it's separated, we can conditionally get results using a separate method, and it could be configured.

**Comment by abrthel** [comment]
@solnic how would you envision configuring if results are returned or not? 

I imagine, you'd want to be able to do something like:
`relation.command(:create, option_name: :value)`

but what would you call the option?

**Comment by solnic** [comment]
So, been thinking about this and realized it's something we want to do in rom 5.0 / rom-sql 3.0.

**Comment by paddor** [comment]
@solnic Did something to fix this behavior make it into rom 5.0 / rom-sql 3.0? I'd like to avoid an additional `SELECT` query after a `:create` command.

**Comment by solnic** [comment]
@paddor no, but it's very likely it'll be addressed in rom 6.0.0 and rom-sql 4.0
